appium/api-mocks/apis/google/google-data.ts

```diff
@@ -389,14 +389,28 @@ export class GoogleData {
     const subject = (query?.match(/subject: '([^"]+)'/) || [])[1]?.trim().toLowerCase();
     const threads: GmailThread[] = [];
 
-    for (const thread of this.getMessagesAndDrafts().
-      filter(m => labelIds.length ? (m.labelIds || []).some(l => labelIds.includes(l)) : true).
+    const filteredThreads = this.getMessagesAndDrafts().
+      filter(m => {
+        if (labelIds.length) {
+          const messageLabels = m.labelIds || [];
+          if (messageLabels.includes('TRASH')) {
+            return labelIds.includes('TRASH');
+          } else {
+            return messageLabels.some(l => labelIds.includes(l));
+          }
+        } else {
+          return true;
+        }
```

Comment:
we don't need `else` statements here. just `return` would do a job.
Not necessary to remove `else` though.